### PR TITLE
Use 'implementation' instead of 'compile'

### PR DESCRIPTION
**Pull request by jaredsburrows**
@JakeWharton

### CODE REVIEW

File: coroutines/build.gradle [hunk 1]
```diff
@@ -9,9 +9,10 @@ repositories {
 }
 
 dependencies {
-  compile deps.kotlin.stdLib.jre
-  compile deps.kotlin.coroutines
-  testCompile deps.junit
+  implementation deps.kotlin.stdLib.jre
+  implementation deps.kotlin.coroutines
```
Comment (JakeWharton) [inline coroutines/build.gradle:13]: Both should be `api`

File: multiplatform/common/build.gradle [hunk 2]
```diff
@@ -1,7 +1,8 @@
 apply plugin: 'org.jetbrains.kotlin.platform.common'
 
 dependencies {
-  compile deps.kotlin.stdLib.common
-  testCompile deps.kotlin.test.annotations
-  testCompile deps.kotlin.test.common
+  implementation deps.kotlin.stdLib.common
```
Comment (JakeWharton) [inline multiplatform/common/build.gradle:4]: `api`

File: multiplatform/jre/build.gradle [hunk 3]
```diff
@@ -7,8 +7,10 @@ archivesBaseName = 'reagent'
 
 dependencies {
   expectedBy project(':multiplatform:common')
-  compile deps.kotlin.stdLib.jre
-  testCompile deps.kotlin.test.jre
+
+  implementation deps.kotlin.stdLib.jre
```
Comment (JakeWharton) [inline multiplatform/jre/build.gradle:11]: `api`

File: multiplatform/js/build.gradle [hunk 4]
```diff
@@ -5,8 +5,10 @@ archivesBaseName = 'reagent'
 
 dependencies {
   expectedBy project(':multiplatform:common')
-  compile deps.kotlin.stdLib.js
-  testCompile deps.kotlin.test.js
+
+  implementation deps.kotlin.stdLib.js
```
Comment (JakeWharton) [inline multiplatform/js/build.gradle:9]: `api`

File: multiplatform/jre/build.gradle (hunk 3 above)
Comment (jaredsburrows) [reply multiplatform/jre/build.gradle]: Good point.

**Comment by jaredsburrows** [comment]
@JakeWharton Without applying `apply plugin: "java-library"`, I cannot use `api`. See:
```
$ gradlew build

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/<>/repo/Reagent/multiplatform/common/build.gradle' line: 4

* What went wrong:
A problem occurred evaluating project ':multiplatform:common'.
> Could not find method api() for arguments [org.jetbrains.kotlin:kotlin-stdlib-common:1.2.0] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

Docs: https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation

**Comment by jaredsburrows** [comment]
@JakeWharton What do you want to here?

**Comment by JakeWharton** [comment]
Fixed this in 049efec for JVM module while I was there, not that it has any real effect since all deps are transitive. The JS plugin doesn't offer the functionality.

**Comment by jaredsburrows** [comment]
Awesome.

**Pull request closed without merging**